bundles/org.openhab.core.voice/src/main/java/org/openhab/core/voice/DialogContext.java

```diff
@@ -137,17 +137,23 @@ public Builder withSink(@Nullable AudioSink sink) {
         }
 
         public Builder withKS(@Nullable KSService service) {
-            this.ks = service;
+            if (service != null) {
+                this.ks = service;
+            }
             return this;
```

Comment:
I wonder if you need to do a check at all here. The field is annotated as `@Nullable` anyway, so setting it to `null` will not change anything. Or is there a place where a valid value could be overwritten?

Reply:
Yes, that is the reason, it contains the default "keyword spotting" service if any, the reason for it to be nullable is that the listenAndAnswer functionality can be used without it.